pandas/io/parsers.py

```diff
@@ -1908,6 +1908,10 @@ def __init__(self, src, **kwds):
                 ]
             else:
                 self.names = list(range(self._reader.table_width))
+        elif self.prefix:
```

Comment:
Actually I think should just do this check higher up in the block as an `if` and not an `elif`. Right now this will still work if you pass `names` in addition to header

Reply:
is it okay now

Reply:
Looks like CI is failing - can you take a look?

Reply:
It says `##[error]The job running on agent Azure Pipelines 20 ran longer than the maximum time of 60 minutes. For more information, see https://go.microsoft.com/fwlink/?linkid=2077134
`